vrx_gz/src/NavigationScoringPlugin.cc

```diff
@@ -389,65 +424,108 @@ void NavigationScoringPlugin::PreUpdate( const sim::UpdateInfo &_info,
   if (this->ScoringPlugin::TaskState() != "running")
     return;
 
-  // Current score
-  this->ScoringPlugin::SetScore(std::min(this->RunningStateDuration(),
-    std::chrono::duration<double>(this->ElapsedTime()).count() +
-    this->NumCollisions() * this->dataPtr->obstaclePenalty) /
-      this->dataPtr->numGates);
+  // Compute collisions.
+  auto newCollisions = this->NumCollisions() - this->dataPtr->prevCollisions;
+  if (newCollisions > 0)
+  {
+    auto newScore =
+      this->Score() - newCollisions * this->dataPtr->obstaclePenalty;
+    newScore = std::max(0.0, newScore);
```

Comment:
This means scores can never be negative. I think it's a good rule, so I will add this to the documentation.